src/test/java/teammates/test/cases/BaseTestCaseWithDatastoreAccess.java

```diff
@@ -0,0 +1,303 @@
+package teammates.test.cases;
+
+import java.util.Map;
+
+import teammates.common.datatransfer.AccountAttributes;
+import teammates.common.datatransfer.CommentAttributes;
+import teammates.common.datatransfer.CourseAttributes;
+import teammates.common.datatransfer.DataBundle;
+import teammates.common.datatransfer.EntityAttributes;
+import teammates.common.datatransfer.FeedbackQuestionAttributes;
+import teammates.common.datatransfer.FeedbackResponseAttributes;
+import teammates.common.datatransfer.FeedbackResponseCommentAttributes;
+import teammates.common.datatransfer.FeedbackSessionAttributes;
+import teammates.common.datatransfer.InstructorAttributes;
+import teammates.common.datatransfer.StudentAttributes;
+import teammates.common.datatransfer.StudentProfileAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.JsonUtils;
+import teammates.common.util.StringHelper;
+import teammates.common.util.ThreadHelper;
+import teammates.test.driver.BackDoor;
+
+/**
+ * Base class for all test cases which are allowed to access the Datastore via {@link BackDoor}.
+ */
+@SuppressWarnings("PMD.TestClassWithoutTestCases") // false positive by PMD
+public class BaseTestCaseWithDatastoreAccess extends BaseTestCase {
```

Comment:
I'm going to make this class `abstract` as well (as it should have been). It also removed the need for the `@SuppressWarnings`.